pkg/vcd/volume.go

```diff
@@ -0,0 +1,160 @@
+package vcd
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"github.com/vmware/cloud-provider-for-cloud-director/pkg/vcdsdk"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func GetDiskRecordsOfClusterByDescription(vcdClient *vcdsdk.Client, ClusterId string) ([]*types.DiskRecordType, error) {
+	filter := "description==" + url.QueryEscape(ClusterId)
```

Comment:
has this always been there? I've checked the records using the cd-cli and it's really there for some disks. Was I blind or you've added some smartness also somewhere else (CSI maybe)?

```
  {
    "Xmlns": "",
    "HREF": "https://vmware.ikoula.com/api/disk/93caa223-c315-4994-82bd-02e8e883ac26",
    "Id": "",
    "Type": "",
    "Name": "pvc-3e1ec10b-fcaa-4b16-97ab-693dcf0ed471",
    "Vdc": "https://vmware.ikoula.com/api/vdc/2ad2288b-f5fd-4410-bbc5-9e470bf4ae2e",
    "SizeMb": 1024,
    "Iops": 0,
    "Encrypted": false,
    "UUID": "6000c294-521a-7d6f-4cc4-18ab46456330",
    "DataStore": "",
    "DataStoreName": "",
    "OwnerName": "xaviertn",
    "VdcName": "vDC 73640",
    "Task": "",
    "StorageProfile": "https://vmware.ikoula.com/api/vdcStorageProfile/69a217fb-7f8a-45fb-aea9-860bbcff9131",
    "StorageProfileName": "vSAN-RAID-5",
    "Status": "RESOLVED",
    "BusType": "6",
    "BusSubType": "pvscsi",
    "BusTypeDesc": "Paravirtual (SCSI)",
    "AttachedVmCount": 0,
    "SharingType": "None",
    "IsAttached": false,
    "IsShareable": false,
    "Description": "NO_RDE_4f0dcd2a-18b6-4ccc-9870-6af0d53381e1",
    "Link": null,
    "Metadata": null
  }
```

Reply:
meh, scratch that ^ I didn't read the PR description properly :D